TeaDiary/Scripts/App/TeaCommands.js

```diff
@@ -0,0 +1,46 @@
+﻿function TeaCommands() {
+    var self = this;
+    self.delete = (id, successCallback, errorCallback) => {
+        popupConfirmation(
+            {
+                 body: "Are you sure you want to delete this tea? You cannot undo this action"
+            },
+            () => apiCommands.deleteTea(id, successCallback, errorCallback)
+        );
+    }
+
+    function popupConfirmation(content, confirmCallback, rejectCallback) {
```

Comment:
It's a noticeable violation of the single responsibility principle. Deletion and Confirmation are separate concerns. In the future, you may want to use this Confirmation logic in a different place, and the Deletion logic is strongly bound to My Teas feature. Let me know how you'll go about fixing this.

Reply:
Well, move confirmation to different class and just use instance of it in deletion

Reply:
It's a good solution